modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -450,6 +450,34 @@ TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, DISABLED_regression)
 
 #endif // UPDATE_QRCODE_TEST_DATA
 
+TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, regression_22205)
+{
+    cv::Ptr<cv::QRCodeEncoder> encoder = nullptr;
+    std::string input_data = "the quick brown fox jumps over the lazy dog";
+
+    std::vector<cv::QRCodeEncoder::EncodeMode> encode_modes = {
+        cv::QRCodeEncoder::EncodeMode::MODE_NUMERIC,
+        cv::QRCodeEncoder::EncodeMode::MODE_ALPHANUMERIC,
+        cv::QRCodeEncoder::EncodeMode::MODE_BYTE};
+
+    std::vector<cv::Mat> result_qrcodes;
+
+    cv::QRCodeEncoder::Params params;
+
+    for(const auto& encode_mode : encode_modes)
+    {
+        params.mode = encode_mode;
+        for(size_t struct_num = 2; struct_num < 5; ++struct_num)
+        {
+            params.structure_number = static_cast<int>(struct_num);
+            encoder = cv::QRCodeEncoder::create(params);
```

Comment:
RAII is preferable here:
```
cv::Ptr<cv::QRCodeEncoder> encoder = cv::QRCodeEncoder::create(params);
```

Reply:
It is not about RAII only. It is mostly about declaration of variables near their usage.
Prefer to keep variables scope minimal.